protocol/x/clob/e2e/order_removal_test.go

```diff
@@ -361,6 +365,362 @@ func TestConditionalOrderRemoval(t *testing.T) {
+				constants.LongTermOrder_Carl_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10,
+			},
+			// Proposer tries to remove first order but cannot because subaccount would still be well
+			// collateralized if order was fully filled.
+			msgProposedOperations: &clobtypes.MsgProposedOperations{
+				OperationsQueue: []clobtypes.OperationRaw{
+					clobtestutils.NewOrderRemovalOperationRaw(
+						constants.LongTermOrder_Carl_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10.OrderId,
+						clobtypes.OrderRemoval_REMOVAL_REASON_UNDERCOLLATERALIZED,
+					),
+				},
+			},
+			expectedErrType: clobtypes.ErrInvalidOrderRemoval,
+			expectedErr:     "Order passes collateralization check",
+		},
+		// Re-enable when reduce-only orders are re-enabled.
+		// "invalid proposal: valid reduce-only order cannot be removed": {
+		// 	subaccounts: []satypes.Subaccount{
+		// 		constants.Carl_Num0_1BTC_Short,
+		// 	},
+		// 	orders: []clobtypes.Order{
+		// 		constants.LongTermOrder_Carl_Num0_Id2_Clob0_Buy10_Price35_GTB20_RO,
+		// 	},
+		// 	msgProposedOperations: &clobtypes.MsgProposedOperations{
+		// 		OperationsQueue: []clobtypes.OperationRaw{
+		// 			clobtestutils.NewOrderRemovalOperationRaw(
+		// 				constants.LongTermOrder_Carl_Num0_Id2_Clob0_Buy10_Price35_GTB20_RO.OrderId,
+		// 				clobtypes.OrderRemoval_REMOVAL_REASON_INVALID_REDUCE_ONLY,
+		// 			),
+		// 		},
+		// 	},
+		// 	expectedErr: "Order fill must increase position size or change side",
+		// },
+		"invalid proposal: non reduce-only order may not be removed with reduce-only reason": {
+			subaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_1BTC_Short,
+			},
+			orders: []clobtypes.Order{
+				constants.LongTermOrder_Carl_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10,
+			},
+			msgProposedOperations: &clobtypes.MsgProposedOperations{
+				OperationsQueue: []clobtypes.OperationRaw{
+					clobtestutils.NewOrderRemovalOperationRaw(
+						constants.LongTermOrder_Carl_Num0_Id0_Clob0_Buy1BTC_Price50000_GTBT10.OrderId,
+						clobtypes.OrderRemoval_REMOVAL_REASON_INVALID_REDUCE_ONLY,
+					),
+				},
+			},
+			expectedErrType: clobtypes.ErrInvalidOrderRemoval,
+			expectedErr:     "Order must be reduce only",
+		},
+		"invalid proposal: conditional fok order cannot be removed when untriggered": {
+			subaccounts: []satypes.Subaccount{
+				constants.Carl_Num0_10000USD,
+			},
+			orders: []clobtypes.Order{
+				constants.ConditionalOrder_Carl_Num0_Id0_Clob0_Buy05BTC_Price50000_GTBT10_SL_50003_FOK,
+			},
+			msgProposedOperations: &clobtypes.MsgProposedOperations{
+				OperationsQueue: []clobtypes.OperationRaw{
+					clobtestutils.NewOrderRemovalOperationRaw(
+						constants.ConditionalOrder_Carl_Num0_Id0_Clob0_Buy05BTC_Price50000_GTBT10_SL_50003_FOK.OrderId,
+						clobtypes.OrderRemoval_REMOVAL_REASON_CONDITIONAL_FOK_COULD_NOT_BE_FULLY_FILLED,
+					),
+				},
+			},
+			expectedErrType: clobtypes.ErrStatefulOrderDoesNotExist,
+			expectedErr:     "does not exist in triggered conditional state",
```

Comment:
meganit: maybe just "conditional order is untriggered"? errors should be easy to understand even for someone without knowledge about the protocol

Reply:
This error message is coming from the stateful validation jonathan just implemented where FetchOrderFromOrderId returns an error if the order cannot be found in triggered state. This function doesn't differentiate between: not in state and in state but untriggered, all it cares about is whether or not it is found in triggered state. Think that is most descriptive and accurate for now